### DISCUSSION TITLE
Proper way of handling pull requests in case of check failure

### DISCUSSION THREAD

**Discussion by viking1304** [Homebrew/discussions#2259] [Writing Formulae/Casks] [answered]
What is preferred way how to handle pull requests in case of check failure?

I wanted to submit a simple pull request for samsung-dex update. I used brew bump-cask-pr and provided new version and new sha256.

CI reported an error, so I fixed that error. Then I got another error. And another.

Finally all checks were fine, but my pull request was a result of 4 small commits instead of one decent commit (because I didn't noticed that dmg name was changed in this new version of Samsung Dex).

Is this fine, or next time I should revoke my pull request if I notice a failure, do all the changes and submit new pull request?

**Comment by carlocab** [comment, marked as answer]
Multiple commits at Homebrew/cask are fine because they can easily be squashed upon merging. At Homebrew/core, they're also easily squashed for version bump or new formula PRs.

You don't need to close your pull request to clean up your PR, though. You can always use `git commit --amend`, `git rebase`, and `git push --force` to amend your commits and squash them together.
